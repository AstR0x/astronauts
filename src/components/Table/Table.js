import React from 'react';

import { formatDate } from '../../utils';

import { ReactComponent as DeleteIcon } from '../../assets/icons/delete.svg';

import styles from './Table.module.css';

const Table = ({ astronauts, onDelete }) => (
  <table className={styles.table}>
    <thead className={styles.head}>
      <tr className={styles.row}>
        <th className={styles.cell}>Имя</th>
        <th className={styles.cell}>Дата</th>
        <th className={styles.cell}>Количество дней</th>
        <th className={styles.cell}>Название миссии</th>
        <th className={styles.cell}>Повторные полёты</th>
      </tr>
    </thead>
    <tbody className={styles.body}>
      {astronauts.map(astronaut => (
        <tr className={styles.row} key={astronaut.name}>
          <td className={styles.cell}>{astronaut.name}</td>
          <td className={styles.cell}>{formatDate(astronaut.date)}</td>
          <td className={styles.cell}>{astronaut.days}</td>
          <td className={styles.cell}>{astronaut.mission}</td>
          <td className={styles.cell}>{astronaut.isMultiple ? 'Да' : 'Нет'}</td>
          <button
            onClick={onDelete.bind(null, astronaut.id)}
            className={styles.deleteButton}
            type="button"
          >
            <DeleteIcon />
          </button>
        </tr>
      ))}
    </tbody>
  </table>
);

export { Table };
