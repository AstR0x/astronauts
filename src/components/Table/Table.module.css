.table {
    display: block;
    margin: 8px auto;
    box-shadow: rgba(0, 0, 0, 0.4) 0 1px 7px -1px;
}

.head {
    height: 50px;
    display: block;
    background-color: #007bff;
    color: #fff;
    font-family: sans-serif;
}

.sortedCell {
    cursor: pointer;
}

.icon {
    width: 20px;
    height: 20px;
}

.row {
    min-height: 50px;
    padding: 10px 0;
    display: flex;
    align-items: center;
    border-bottom: 1px solid #eee;
    position: relative;
}

.deleteButton {
    width: 35px;
    height: 35px;
    position: absolute;
    top: 6px;
    right: 10px;
    padding-top: 4px;
    background-color: #fff;
    border: 1px solid #ccc;
    border-radius: 4px;
    outline: none;
}

.deleteButton:hover {
    background-color: #f0f0f0;
}

.deleteButton:focus {
    border: 1px solid #007bff;
}

.cell {
    width: 20%;
    text-align: center;
}

.body {
    display: block;
}

@media (max-width: 414px) {
    .row {
        padding: 0 4px;
    }

    .deleteButton {
        display: none;
    }
}
