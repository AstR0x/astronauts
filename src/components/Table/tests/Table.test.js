import React from 'react';
import renderer from 'react-test-renderer';

import { Table } from 'components/Table';

const astronauts = [
  {
    id: 1,
    name: 'Sigmund JÃ¤hn',
    date: 272926800,
    days: 7,
    mission: 'Sojus 31 / Sojus 29',
    isMultiple: false,
  },
  {
    id: 2,
    name: 'Ulf Merbold',
    date: 438814800,
    days: 10,
    mission: 'STS-9',
    isMultiple: true,
  },
  {
    id: 3,
    name: 'Reinhard Furrer',
    date: 499467600,
    days: 7,
    mission: 'STS-61-A (D1)',
    isMultiple: false,
  }];

describe('component Table should', () => {
  const component = renderer.create(
    <Table
      astronauts={astronauts}
      onSort={() => ({})}
      onDelete={() => ({})}
      sortField={() => ({})}
      sortDirection={() => ({})}
    />,
  );

  const tree = component.toJSON();

  it('correct render', () => {
    expect(tree).toMatchSnapshot();
  });
});
