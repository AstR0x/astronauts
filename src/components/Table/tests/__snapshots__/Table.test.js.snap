// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Table should correct render 1`] = `
<table
  className="table"
>
  <thead
    className="head"
  >
    <tr
      className="row"
    >
      <th
        className="cell sortedCell"
        onClick={[Function]}
      >
        Имя
         
      </th>
      <th
        className="cell sortedCell"
        onClick={[Function]}
      >
        Дата полёта
         
      </th>
      <th
        className="cell sortedCell"
        onClick={[Function]}
      >
        Дни в космосе
         
      </th>
      <th
        className="cell sortedCell"
        onClick={[Function]}
      >
        Название миссии
         
      </th>
      <th
        className="cell sortedCell"
        onClick={[Function]}
      >
        Повторные полёты
         
      </th>
      <th />
    </tr>
  </thead>
  <tbody
    className="body"
  >
    <tr
      className="row"
    >
      <td
        className="cell"
      >
        Sigmund Jähn
      </td>
      <td
        className="cell"
      >
        25.08.1978
      </td>
      <td
        className="cell"
      >
        7
      </td>
      <td
        className="cell"
      >
        Sojus 31 / Sojus 29
      </td>
      <td
        className="cell"
      >
        Нет
      </td>
      <td>
        <button
          className="deleteButton"
          onClick={[Function]}
          type="button"
        >
          <svg>
            delete.svg
          </svg>
        </button>
      </td>
    </tr>
    <tr
      className="row"
    >
      <td
        className="cell"
      >
        Ulf Merbold
      </td>
      <td
        className="cell"
      >
        27.11.1983
      </td>
      <td
        className="cell"
      >
        10
      </td>
      <td
        className="cell"
      >
        STS-9
      </td>
      <td
        className="cell"
      >
        Да
      </td>
      <td>
        <button
          className="deleteButton"
          onClick={[Function]}
          type="button"
        >
          <svg>
            delete.svg
          </svg>
        </button>
      </td>
    </tr>
    <tr
      className="row"
    >
      <td
        className="cell"
      >
        Reinhard Furrer
      </td>
      <td
        className="cell"
      >
        29.10.1985
      </td>
      <td
        className="cell"
      >
        7
      </td>
      <td
        className="cell"
      >
        STS-61-A (D1)
      </td>
      <td
        className="cell"
      >
        Нет
      </td>
      <td>
        <button
          className="deleteButton"
          onClick={[Function]}
          type="button"
        >
          <svg>
            delete.svg
          </svg>
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
